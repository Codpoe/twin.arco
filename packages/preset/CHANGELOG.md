# @twin.arco/preset

## 0.1.1

### Patch Changes

- c6d7a53: Fix: wrong devpendence version

## 0.1.0

### Minor Changes

- Feature: first release
