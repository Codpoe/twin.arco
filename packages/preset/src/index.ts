import { windi } from '@twin.arco/dark-mode';
import { createRgba, createRgb, defaultKeyFormatter } from './utils';

export * from './utils';

export interface PresetOptions {
  keyFormatter?: typeof defaultKeyFormatter;
  enableOpacity?: boolean;
  darkMode?: 'windicss' | 'tailwindcss' | false;
}

export default function ({
  keyFormatter = defaultKeyFormatter,
  enableOpacity = true,
  darkMode = false,
}: PresetOptions = {}) {
  const plugins = darkMode === 'windicss' ? [windi] : undefined;
  return {
    darkMode: plugins?.length ? false : (undefined as false | undefined),
    plugins,
    theme: {
      extend: {
        colors: {
          ...createRgb(keyFormatter),
          ...createRgba(keyFormatter, undefined, enableOpacity),
        },
      },
    },
  };
}
