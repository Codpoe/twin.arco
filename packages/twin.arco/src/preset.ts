import preset from '@twin.arco/preset';

export default preset;
