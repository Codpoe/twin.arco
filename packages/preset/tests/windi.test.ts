import { Processor } from 'windicss/lib/index';
import preset from '@/index';

describe('Twin.arco preset for windi', () => {
  test('WindiCSS preset', () => {
    const processor = new Processor({
      presets: [preset()],
    });
    const result = processor.interpret(
      'text-red-1 bg-red-1 border-red-1 text-arco-text-1 bg-arco-bg-1 border-arco-border-1',
    );
    expect(result.ignored.length).toBe(0);
    expect(result.styleSheet.build()).toMatchSnapshot();
  });

  test('WindiCSS disable opacity', () => {
    const processor = new Processor({
      presets: [preset({ enableOpacity: false })],
    });
    const result = processor.interpret('text-red-1 text-opacity-50');
    expect(result.ignored.length).toBe(0);
    expect(result.styleSheet.build()).toMatchSnapshot();
  });

  test('WindiCSS dark mode', () => {
    const processor = new Processor({
      presets: [preset({ darkMode: 'windicss' })],
    });
    const result = processor.interpret('text-red-1 dark:text-red-10');
    expect(result.ignored.length).toBe(0);
    expect(result.styleSheet.build()).toMatchSnapshot();
  });
});
