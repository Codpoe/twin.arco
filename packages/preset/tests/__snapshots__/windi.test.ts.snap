// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twin.arco preset for windi WindiCSS dark mode 1`] = `
".text-red-1 {
  --tw-text-opacity: 1;
  color: rgba(var(--red-1), var(--tw-text-opacity, 1));
}
body[arco-theme=dark] .dark\\\\:text-red-10 {
  --tw-text-opacity: 1;
  color: rgba(var(--red-10), var(--tw-text-opacity, 1));
}"
`;

exports[`Twin.arco preset for windi WindiCSS disable opacity 1`] = `
".text-red-1 {
  color: rgb(var(--red-1));
}
.text-opacity-50 {
  --tw-text-opacity: 0.5;
}"
`;

exports[`Twin.arco preset for windi WindiCSS preset 1`] = `
".bg-red-1 {
  --tw-bg-opacity: 1;
  background-color: rgba(var(--red-1), var(--tw-bg-opacity, 1));
}
.bg-arco-bg-1 {
  background-color: var(--color-bg-1);
}
.border-red-1 {
  --tw-border-opacity: 1;
  border-color: rgba(var(--red-1), var(--tw-border-opacity, 1));
}
.border-arco-border-1 {
  border-color: var(--color-border-1);
}
.text-red-1 {
  --tw-text-opacity: 1;
  color: rgba(var(--red-1), var(--tw-text-opacity, 1));
}
.text-arco-text-1 {
  color: var(--color-text-1);
}"
`;
